ipykernel/compiler.py

```diff
@@ -36,11 +37,35 @@ def murmur2_x86(data, seed):
 
     return h
 
+convert_to_long_pathname = lambda filename:filename
+
+if sys.platform == 'win32':
+    try:
+        import ctypes
+        from ctypes.wintypes import MAX_PATH, LPCWSTR, LPWSTR, DWORD
+
+        _GetLongPathName = ctypes.windll.kernel32.GetLongPathNameW
+        _GetLongPathName.argtypes = [LPCWSTR, LPWSTR, DWORD]
+        _GetLongPathName.restype = DWORD
+
+        def _convert_to_long_pathname(filename):
+            buf = ctypes.create_unicode_buffer(MAX_PATH)
+            rv = _GetLongPathName(filename, buf, MAX_PATH)
+            if rv != 0 and rv <= MAX_PATH:
+                filename = buf.value
+            return filename
+
+        # test that it works
+        _convert_to_long_pathname(__file__)
```

Comment:
Can you please add this as a test in `test_debugger.py` instead of inline here?

Reply:
I can add it there to test as well. The intent with adding it here is to catch any errors with e.g. DLL references early so we can fail just once up front and fallback to not normalizing, rather than failing every time we get the path below.